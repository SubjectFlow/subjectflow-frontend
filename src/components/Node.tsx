import React from "react";
import "../styles/Node.css";
import { Colour, Point } from "../Types";

type NodeProps = {
  mainColour: Colour;
  accentColour: Colour;
  position: Point;
  title: string;
  code?: string;
  type: string;
  major?: boolean;
};

function Node(props: NodeProps) {
  return (
    <div
      className="node"
      style={{
        backgroundColor: props.mainColour,
        color: props.accentColour,
        bottom: props.position.y,
        left: props.position.x
      }}
    >
      <div
        className="node__title"
        style={props.major ? { overflow: "visible", whiteSpace: "normal" } : {}}
      >
        {props.title}
      </div>
      {!props.major && (
        <div className="node__code">
          {props.major ? " " : props.code ? props.code : " "}
        </div>
      )}
      <div className="node__type">{props.type}</div>
    </div>
  );
}

export default Node;
