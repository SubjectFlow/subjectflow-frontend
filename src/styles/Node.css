.node {
  border-radius: 5px;
  font-family: "Inter", sans-serif;
  position: absolute;
  text-align: left;
  padding: 4px 5px 3px 5px;
  width: 230px;
  height: min-content;
}

.node__title {
  font-weight: normal;
  font-size: 13pt;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.node__code {
  font-family: "Montserrat", sans-serif;
  font-weight: 200;
  font-size: 10pt;
  position: relative;
  top: -1px;
}

.node__type {
  font-weight: lighter;
  font-size: 10pt;
  text-align: right;
}
