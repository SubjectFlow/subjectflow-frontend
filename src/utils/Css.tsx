export const getCssVarInt = (name: string): number => {
  return parseInt(getComputedStyle(document.documentElement).getPropertyValue(name));
};
