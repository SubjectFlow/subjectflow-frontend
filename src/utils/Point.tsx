import { Point } from "../Types";

const origin = { x: 0, y: 0 };

function dist(p1: Point, p2: Point) {
  let dx = p2.x - p1.x;
  let dy = p2.y - p2.y;
  return Math.sqrt(dx * dx + dy * dy);
}

function length(p: Point) {
  return dist(origin, p);
}

function distSq(p1: Point, p2: Point) {
  let dx = p2.x - p1.x;
  let dy = p2.y - p2.y;
  return dx * dx + dy * dy;
}

function add(p1: Point, p2: Point) {
  return {
    x: p1.x + p2.x,
    y: p1.y + p2.y
  };
}

function flip(p: Point) {
  return {
    x: -p.x,
    y: -p.y
  };
}

function normalise(p: Point) {
  let len = length(p);
  return {
    x: p.x / len,
    y: p.y / len
  };
}

export const PointUtils = {
  distSq: distSq,
  dist: dist,
  add: add,
  length: length,
  flip: flip,
  normalise: normalise
};
