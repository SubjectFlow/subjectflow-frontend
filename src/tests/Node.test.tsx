import React from "react";
import renderer from "react-test-renderer";
import Node from "../components/Node";
import { Point } from "../utils/Point"

test("node renders test title", () => {
  const component = renderer.create(
    <Node
      onClick={(x: number) => {}}
      faded={false}
      mainColour="#004369"
      accentColour="#EDEDED"
      edgeListID={0}
      position={new Point(400, 370)}
      content={{
        id: "a",
        name: "Artificial Intelligence",
        code: "COMP30024",
        type: "ELECTIVE SCIENCE"
      }}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
