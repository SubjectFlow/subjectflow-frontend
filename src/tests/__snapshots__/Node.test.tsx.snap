// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`node renders test title 1`] = `
<div
  className="node"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#004369",
      "bottom": 370,
      "color": "#EDEDED",
      "left": 400,
    }
  }
>
  <div
    className="noselect node__title"
  >
    Artificial Intelligence
  </div>
  <div
    className="noselect node__code"
  >
    COMP30024
  </div>
  <div
    className="noselect node__type"
  >
    ELECTIVE SCIENCE
  </div>
</div>
`;
